### PR TITLE
Added option of supplying directory to rst-lint

### PR DESCRIPTION
**Pull request by dhruvsomani**
This commit adds the feature of supplying a directory to rst-lint. The program *recurses* (so all files within subdirectories will be linted too) to find all the `.rst` files in the directory, and lints the files.

Also added tests for the same.

Closes https://github.com/twolfson/restructuredtext-lint/issues/38

### CODE REVIEW

File: restructuredtext_lint/cli.py [hunk 1]
```diff
@@ -37,8 +37,21 @@ def _main(filepaths, format=DEFAULT_FORMAT, stream=sys.stdout, encoding=None, le
...
+    for path in filepaths:
+        # Check if the given path is a file or a directory
+        if (os.path.isfile(path)):
+            paths.append(path)
+        else:
+            for file in os.listdir(path):
+                fname, fext = os.path.splitext(file)
+                if fext == '.rst':
+                    paths.append(os.path.join(path, file))
+                elif fext == '':
```
Comment (twolfson) [inline restructuredtext_lint/cli.py]: Unfortunately, this is going to fail for directories that contain a dot in them (e.g. `foo.app`). We probably need to iterate recursively (i.e. going through the same `.isfile` process -- the easiest way to do this should be `os.walk`)

https://docs.python.org/2/library/os.html#os.walk

File: restructuredtext_lint/test/test.py [hunk 2]
```diff
@@ -169,6 +170,12 @@ def test_rst_lint_correct_file(self):
         output = str(raw_output)
         self.assertEqual(output, '')
 
+    def test_rst_lint_folder(self):
+        """The `rst-lint` command should print errors in the files inside the folder."""
+        output = subprocess.check_output((sys.executable, rst_lint_path, f_rst), universal_newlines=True)
+        output = str(output)
+        self.assertEqual(output, '')
```
Comment (twolfson) [inline restructuredtext_lint/test/test.py]: This doesn't seem to be asserting that any files in the folder are failing. Can we verify that it indeed checks the file contents?

**Comment by twolfson** [comment]
Looks good but it seems like tests could use some work and we missed an edge case for traversal

**Comment by dhruvsomani** [comment]
@twolfson Made the changes.

**Comment by twolfson** [comment]
This looks good at a glance. I'm unsure I'll be able to fully land it tonight but I'll do it by the end of the week

**Comment by twolfson** [comment]
This has been released in `1.2.0`. Thanks for your help on making this feature happen =D
